syncmaster/scheduler/transfer_job_manager.py

```diff
@@ -47,6 +49,18 @@ def update_jobs(self, transfers: list[Transfer]) -> None:
                     args=(transfer.id,),
                 )
 
+    async def remove_orphan_jobs(self) -> None:
+        all_jobs = self.scheduler.get_jobs()
+        settings = self.settings
+
+        async with get_async_session(settings) as session:
+            transfer_ids = {t.id for t in (await session.execute(select(Transfer))).scalars().all()}
+
+        for job in all_jobs:
+            transfer_id = int(job.id)
+            if transfer_id not in transfer_ids:
+                self.scheduler.remove_job(job.id)
```

Comment:
Could we pass list of known job_ids to an SQL query to avoid fullscanning the entire table?